Creating Estimate Section

alaasaedy opened the pull request.

alaasaedy committed:
Creating Estimate Section

saifaldin commented on frontend/src/components/EstimateSection/index.jsx:
```diff
@@ -0,0 +1,23 @@
+
+const EstimateSection = () => {
+  return (
+    <div className="estimate__section_bg py-32">
```
please use section instead of div, to apply the global section style. Also for semantics.

MAlngaawy reviewed (changes requested):
**Changes request**
- please change the ( have any projects in mind ) font family to be `PT serif`
- please make py = 6rem instead of 8 rem to match the design height
- please change the color of the font to `c-200` like the design
- please try to use inline style to add the background image instead of using a custom CSS file
- and add the margin Y to the whole section

**and finally try to fetch the titles and button data from the server API**

alaasaedy commented:
Thanks for your review.
about the padding in the second point i can't find any py-8 in code. please point to the line number if available.
and for the data fetching I'm not aware yet how to do that.

alaasaedy committed:
change section font family, colors.
use semantic markup section to apply the global styling for section tag.
use inline styling to add the bg image instead of using custom css file.

alaasaedy committed:
remove unnecessary className attribute

alaasaedy committed:
addiing data in DB to use effect

alaasaedy committed:
adding section data to DB and use hooks to fetch it to the component then using it

MAlngaawy commented on backend/db.json:
```diff
@@ -7,5 +7,10 @@
+  "estimatesection": {
+    "subTitle": "Estimate your Project!",
+    "sectionTitle": "Have Any Project On Mind ?",
+    "btnLabel": "get in touch"
```
I think You get confused about the title and subtitle.
- the `API` should be like 
"title": "Have Any Project On Mind?" ==> and just title .. not `sectionTitle`
"subTitle": "Estimate your Project!"
"button": {
"title": "Get Touch",
"url": "/"
}
- please put them at the right place, don't reflect the titles

MAlngaawy commented on frontend/src/components/EstimateSection/index.jsx:
```diff
@@ -0,0 +1,44 @@
+    API('/estimatesection').then(({ data: estimatesection }) => {
+      setData(estimatesection);
+    });
+  });
```
no need to do this `({data: estimatesection})`, just keep it `data` and pass it to the state like this `setData(data)`

**and please add the brackets `[]` before the useEffect closing**

> because it caused to infinity rerender

useEffect should be like this 
`useEffect(() => { //your code here }, []);`

alaasaedy committed:
add conditional rendering to the jsx return to check the data is returned successfully

destructuring returened data
decress heading my

alaasaedy committed:
Merge branch 'master' of https://github.com/m3ntorship/personal-resume into estimate-section

MAlngaawy reviewed (approved):
LGTM
